bignumber.js

```diff
@@ -2877,6 +2865,47 @@
   }
 
 
+  function createAlphabet(alphabet) {
+    var i, j, c, d, caseSensitive, lowerAlphabet;
+
+    // check if alphabet is case-sensitive
+    for (i = 0; i < alphabet.length; i++) {
+      c = alphabet.charAt(i);
+      for (j = 0; j < i; j++) {
+        d = alphabet.charAt(j);
+        if (c.toLowerCase() === d.toLowerCase()) {
+          caseSensitive = true;
+        }
+      }
+    }
+
+    if (!caseSensitive) {
+      lowerAlphabet = alphabet.toLowerCase();
+    }
+
+    function charAt(i) {
+      return alphabet.charAt(i);
+    }
+
+    // case-insensitive version of indexOf
+    function charIndex(c, b) {

```

Comment:
The definition of the `charIndex` function could depend on the value of `caseSensitive`, rather than checking its value every call.